app/screens/create_channel/create_channel.js

```diff
@@ -73,35 +76,32 @@ export default class CreateChannel extends PureComponent {
         this.emitCanCreateChannel(false);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.theme !== nextProps.theme) {
-            setNavigatorStyles(this.props.componentId, nextProps.theme);
+    static getDerivedStateFromProps(props, state) {
```

Comment:
This function also has a lot that needs to be moved out into `componentDidUpdate`

Reply:
@hmhealey is it ok to call setState() in componentDidUpdate()?

Reply:
No, we're avoiding doing that because it causes a second re-render when that happens. Instead, you can usually modify the state in `getDerivedStateFromProps` and then check to see that that state changed in `componentDidUpdate` to trigger other things like `setNavigatorStyles`

Reply:
inorder to update the state in the `getDerivedStateFromProps` can i include the `createChannelRequest` as a part of state?

Reply:
Yes you can. But if you can get rid of that prop entirely that would be short of amazing!!

Reply:
Although to be fair is not part of this help wanted issue

Reply:
Yeah, ideally we wouldn't need `getDerivedStateFromProps` at all (see [this blog post](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html) from the React team), but if we do need it here, you can definitely add that to the state.